### ISSUE TITLE
Publish routes-compiler for Scala 2.11

### ISSUE THREAD

**Issue by benmccann** [playframework/playframework#5290]
I'd like to use the `routes-compiler` project in my Play project to introspect the routes in my project. Unfortunately, I can't do this since it's published only for Scala 2.10.

See https://github.com/swagger-api/swagger-play/pull/20 for where someone is looking to use the routes-compiler, but had to copy/paste all the code into their Play plugin

**Comment by benmccann** [comment]
The `RoutesCompilerProject` is declared here: https://github.com/playframework/playframework/blob/master/framework/project/Build.scala#L187

It seems like `sbt-doge` is used for cross-building: https://github.com/playframework/playframework/blob/master/framework/project/Release.scala#L17

The `build` script sets `-Dsbt.scala.version` to `2.10.5`: https://github.com/playframework/playframework/blob/master/framework/build#L31

I'm not sure at all how to set the build to publish for both Scala 2.10 and 2.11. Hopefully someone a bit more knowledgeable can chime in or help me out with this one.

**Comment by richdougherty** [comment]
At the moment `RoutesCompilerProject` is declared as a `PlaySbtProject`.
```scala
  lazy val RoutesCompilerProject = PlaySbtProject("Routes-Compiler", "routes-compiler")
```
Maybe we can get it to cross build with both 2.10 and 2.11, but maybe that isn't possible with sbt projects.

If it's not possible to cross build an sbt project we might need to split the routes compiler into two projects: one project with the core compiler logic and another project that holds the sbt plugin part. Then it would be easy to cross compile the core compiler logic because it wouldn't mixed in with the sbt plugin code.

@jroper, do you have any suggestions?

**Comment by dwijnand** [comment]
I've found additional pieces to this puzzle over in interplay: https://github.com/playframework/interplay/blob/0a81e81fbc19c4a5cfd562363be133ed7718717d/src/main/scala/interplay/PlayBuildBase.scala#L31-L47

**Comment by richdougherty** [comment]
Should be fixed by #5343.

**Comment by benmccann** [comment]
thanks!!
